src/pages/home/home.tsx

```diff
@@ -1,15 +1,75 @@
 import TopProducts from '../../components/Product/TopProducts';
 import HomeCards from './components/IntroCards/HomeCards';
 import HomeParallax from './components/parallax/homeParallax';
+import ProductMountainSlider from './components/productMountainSlider/productMountainSlider';
 import styles from './home.module.scss';
+import useFetch from '../../hooks/useFetch';
+import { BACKEND_URL } from '../../utils/constants';
+import ProductCard from '../../components/Product/ProductCard';
+
+export interface cardDetails {
+  products: {
+    productImages: { url: string }[];
+    details?: string;
+    id: string;
+    category: string;
+    ratings?: number;
+    price: number;
+    bonus: number;
+    collectionId: string;
+    createdAt?: string;
+    expDate: string;
+    expiredflag: boolean;
+    name: string;
+    quantity: number;
+  }[];
+  message: string;
+  page: number;
+  totalPages: number;
+}
 
 function Home() {
+  const url = `${BACKEND_URL}/products/all?page=1&limit=10`;
+  const { data } = useFetch(url);
+  const instantData: cardDetails | null = data as cardDetails | null;
+  const products = instantData?.products;
+
```

Comment:
Can you create a separate component and not do the fetching and mapping inside this page file?